### ISSUE TITLE
after update sickrage not starting

### ISSUE THREAD

**Issue by jadm557** [SiCKRAGETV/sickrage-issues#2693]
Did an update/ sickrage stopped functioning (no longer starts)
"column indexer_id is not unique"

from the log:
```
ERROR    MAIN :: Fatal error executing query: column indexer_id is not unique
Traceback (most recent call last):
  File "D:\SickRage\sickbeard\db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "D:\SickRage\sickbeard\db.py", line 88, in execute
    return self._execute(query, args)
  File "D:\SickRage\sickbeard\db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
IntegrityError: column indexer_id is not unique
```

On windows envrionment - phython27

-------------------
select *,count(indexer_Id) from Tv_shows group by indexer_Id having count(indexer_Id) > 1
>> 0 rows
>> so if my query is correct: no reused values in the column indexer_id....

**Comment by alekslyse** [comment]
Same issue here on Linux python 2.7.6

Starting SickBeard
Traceback (most recent call last):
  File "SickBeard.py", line 522, in <module>
    SickRage().start()
  File "SickBeard.py", line 295, in start
    sickbeard.initialize(consoleLogging=self.consoleLogging)
  File "/opt/sickbeard/sickbeard/__init__.py", line 1308, in initialize
    db.sanityCheckDatabase(myDB, mainDB.MainSanityCheck)
  File "/opt/sickbeard/sickbeard/db.py", line 254, in sanityCheckDatabase
    sanity_check(connection).check()
  File "/opt/sickbeard/sickbeard/databases/mainDB.py", line 46, in check
    self.convert_tvrage_to_tvdb()
  File "/opt/sickbeard/sickbeard/databases/mainDB.py", line 77, in convert_tvrage_to_tvdb
    (INDEXER_TVDB, mapping[0]['mindexer_id'], tvrage_show['indexer_id'])
  File "/opt/sickbeard/sickbeard/db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/opt/sickbeard/sickbeard/db.py", line 88, in execute
    return self._execute(query, args)
  File "/opt/sickbeard/sickbeard/db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
sqlite3.IntegrityError: UNIQUE constraint failed: tv_shows.indexer_id

**Comment by Plasticjesus690** [comment]
I am also having issues after updating. I get error. 

10:37:00 INFO::CHECKVERSION :: Checking for updates using SOURCE
10:37:00 INFO::MAIN :: Building internal name cache for all shows
10:37:02 ERROR::Thread-13 :: Failed doing webui callback: Traceback (most recent call last):
  File "C:\SickRage-master\sickbeard\webserve.py", line 272, in async_call
    result = function(**kwargs)
  File "C:\SickRage-master\sickbeard\webserve.py", line 728, in index
    return t.render(title="Home", header="Show List", topmenu="home", showlists=showlists, submenu=self.HomeMenu())
  File "C:\SickRage-master\sickbeard\webserve.py", line 163, in render
    return super(PageTemplate, self).render(*args, **kwargs)
  File "C:\SickRage-master\lib\mako\template.py", line 443, in render
    return runtime._render(self, self.callable_, args, data)
  File "C:\SickRage-master\lib\mako\runtime.py", line 803, in _render
    **_kwargs_for_callable(callable_, data))
  File "C:\SickRage-master\lib\mako\runtime.py", line 834, in _render_context
    (inherit, lclcontext) = _populate_self_namespace(context, tmpl)
  File "C:\SickRage-master\lib\mako\runtime.py", line 780, in _populate_self_namespace
    ret = template.module._mako_inherit(template, context)
  File "C:\SickRage-master\cache\mako\SickRage-master\gui\slick\views\home.mako.py", line 69, in _mako_inherit
    return runtime._inherit_from(context, u'/layouts/main.mako', _template_uri)
  File "C:\SickRage-master\lib\mako\runtime.py", line 739, in _inherit_from
    template = _lookup_template(context, uri, calling_uri)
  File "C:\SickRage-master\lib\mako\runtime.py", line 769, in _lookup_template
    return lookup.get_template(uri)
  File "C:\SickRage-master\lib\mako\lookup.py", line 245, in get_template
    return self._load(srcfile, uri)
  File "C:\SickRage-master\lib\mako\lookup.py", line 311, in _load
    **self.template_args)
  File "C:\SickRage-master\lib\mako\template.py", line 321, in __init__
    module = self._compile_from_file(path, filename)
  File "C:\SickRage-master\lib\mako\template.py", line 379, in _compile_from_file
    module = compat.load_module(self.module_id, path)
  File "C:\SickRage-master\lib\mako\compat.py", line 66, in load_module
    return imp.load_source(module_id, path, fp)
  File "C:\SickRage-master\cache\mako\layouts\main.mako.py", line 23, in <module>
    import re, resource
ImportError: No module named resource

**Comment by fieldhof** [comment]
Yep me too, this is the output I got

19:26:24 INFO::MAIN :: Checking for shows with tvrage id's, since tvrage is gone
19:26:24 WARNING::MAIN :: Found 1 shows with TVRage ID', FIXING!
19:26:24 INFO::MAIN :: Mapping Shaun the Sheep Championsheeps to tvdb id 79890
19:26:24 ERROR::MAIN :: Fatal error executing query: UNIQUE constraint failed: tv_shows.indexer_id
Traceback (most recent call last):
  File "/home/user/.sickrage/sickbeard/db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/user/.sickrage/sickbeard/db.py", line 88, in execute
    return self._execute(query, args)
  File "/home/user/.sickrage/sickbeard/db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
IntegrityError: UNIQUE constraint failed: tv_shows.indexer_id
Traceback (most recent call last):
  File "SickBeard.py", line 522, in <module>
    SickRage().start()
  File "SickBeard.py", line 295, in start
    sickbeard.initialize(consoleLogging=self.consoleLogging)
  File "/home/user/.sickrage/sickbeard/__init__.py", line 1308, in initialize
    db.sanityCheckDatabase(myDB, mainDB.MainSanityCheck)
  File "/home/user/.sickrage/sickbeard/db.py", line 254, in sanityCheckDatabase
    sanity_check(connection).check()
  File "/home/user/.sickrage/sickbeard/databases/mainDB.py", line 46, in check
    self.convert_tvrage_to_tvdb()
  File "/home/user/.sickrage/sickbeard/databases/mainDB.py", line 77, in convert_tvrage_to_tvdb
    (INDEXER_TVDB, mapping[0]['mindexer_id'], tvrage_show['indexer_id'])
  File "/home/user/.sickrage/sickbeard/db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/user/.sickrage/sickbeard/db.py", line 88, in execute
    return self._execute(query, args)
  File "/home/user/.sickrage/sickbeard/db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
sqlite3.IntegrityError: UNIQUE constraint failed: tv_shows.indexer_id

**Comment by neoatomic** [comment]
TVRage is down and not coming back up from what we hear. 
That causes some serious prob's that need to get addressed. 
As a temp solution we submitted a fix to convert the TVRage shows to TVDB, but according to your log that failed on your install.

If you cant wait, than rename the sickbeard.db to let SickRage create a new one.

**Comment by Plasticjesus690** [comment]
I have tried a clean fresh copy of the Master build and still receiving issues.

**Comment by ThePOO** [comment]
yep, yep, yep ... broken here.    Waiting for the "fix" ... knowing the fine folks here will make it all happen, waiting patiently .....

**Comment by pavichokche** [comment]
I have the issue as well. Will a regular update be enough to correct it?

**Comment by TheMotto9** [comment]
Althought SR isn't "opening" (displaying in web browser), it seems like SR is still functioning because it's pulling episodes and processing them (even with the "This webpage has a redirect loop ERR_TOO_MANY_REDIRECTS"  error displaying). 

My question is, if I can't display the SR page, how will I push a manual update once this is fixed?

**Comment by madeddie** [comment]
In my case this problem arose because of differences between TVRage and TVDB. In TVRage's DB, Jim Henson's The Storyteller and Jim Henson's The Storyteller: Greek Myths were 2 separate shows with 2 separate show ID's. In TVDB the latter is just another season of the former. Not sure how this can be automagically be fixed for everybody :( I've fixed my problem manually.

Of course checking for this occurrence (multiple TVRage ID's mapping to the same TVDB ID) would help notify this problem, how to proceed from there in a way that's work for everyone, I don't really know.

**Comment by wesh30** [comment]
@jadm557 your query won't return any results, the error is telling you that the indexer_id it's trying to add already exists. look at the error log for the line just above the error that says Mapping 'xxx' to id ###. now check the db and you will see there is already a show with indexer_id ###

**Comment by syrys** [comment]
having the same problem in windows 8.1
````
C:\SickRage>python SickBeard.py
08:56:35 INFO::MAIN :: Checking for shows with tvrage id's, since tvrage is gone

08:56:35 WARNING::MAIN :: Found 6 shows with TVRage ID', FIXING!
08:56:35 INFO::MAIN :: Mapping Marvel's Agents of S.H.I.E.L.D. to tvdb id 263365

08:56:35 ERROR::MAIN :: Fatal error executing query: column indexer_id is not un
ique
Traceback (most recent call last):
  File "C:\SickRage\sickbeard\db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "C:\SickRage\sickbeard\db.py", line 88, in execute
    return self._execute(query, args)
  File "C:\SickRage\sickbeard\db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
IntegrityError: column indexer_id is not unique
Traceback (most recent call last):
  File "SickBeard.py", line 522, in <module>
    SickRage().start()
  File "SickBeard.py", line 295, in start
    sickbeard.initialize(consoleLogging=self.consoleLogging)
  File "C:\SickRage\sickbeard\__init__.py", line 1308, in initialize
    db.sanityCheckDatabase(myDB, mainDB.MainSanityCheck)
  File "C:\SickRage\sickbeard\db.py", line 254, in sanityCheckDatabase
    sanity_check(connection).check()
  File "C:\SickRage\sickbeard\databases\mainDB.py", line 46, in check
    self.convert_tvrage_to_tvdb()
  File "C:\SickRage\sickbeard\databases\mainDB.py", line 77, in convert_tvrage_t
o_tvdb
    (INDEXER_TVDB, mapping[0]['mindexer_id'], tvrage_show['indexer_id'])
  File "C:\SickRage\sickbeard\db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "C:\SickRage\sickbeard\db.py", line 88, in execute
    return self._execute(query, args)
  File "C:\SickRage\sickbeard\db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
sqlite3.IntegrityError: column indexer_id is not unique
````
In my case, its not running in the background, it crashes and dies.

@TheMotto9, as for manually updating without the web UI, in the past when i had similar issues, i was told to do a manual git pull. Assuming you have git installed, you can simply browse to sickrage folder on your console, and type "git pull" (which is what im planning on doing once an update is available). wait till someone else confirms this though.

**Comment by hooliowobbits** [comment]
same here

**Comment by kerbysj** [comment]
Same here

**Comment by wesh30** [comment]
I suspect @TheMotto9 is encountering issue https://github.com/SiCKRAGETV/sickrage-issues/issues/2694.

**Comment by blade57** [comment]
same here

**Comment by TheMotto9** [comment]
@wesh30 @miigotu SR auto-updated and SR's page loads correctly now. But, all my shows are missing cover art and network images are just text now. Is it supposed to be like this due to the issues earlier? Should I do a mass update on all my shows to see if it will "re-downloaded" the missing images?

**Comment by pavichokche** [comment]
Wow, the issue seems to be mostly fixed already. Like @TheMotto9 said everything works fine except for cover art, even some that I had set up myself to point locally (I think I had them set up like that). Either way, thanks so much, SR team !

**Comment by miigotu** [comment]
"Should I do a mass update on all my shows to see if it will "re-downloaded" the missing images?"
Yes.

**Comment by faidwen** [comment]
It doesn't appear to be corrected on Linux boxes with startup.  (as far as I can ascertain)

  File "C:\SickRage\sickbeard\db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
sqlite3.IntegrityError: column indexer_id is not unique

**Comment by miigotu** [comment]
@faidwen thats a different issue, because tvrage was assigning different id's to shows that have a different theme name each season. Its a little complicated because Im not sure how I want to proceed. Should I be moving the files from the one show to the other show's folder or?.?

aka you have one show with just a Season 01 in it, and another show with a season 02 in it, but both are actually the same show.

**Comment by mgaulton** [comment]
@rhiigotu since that would imply some are holding up the process. is there a way to remove them without access to the gui? mine bombs on the first entry lol

Starting SickBeard.
20:34:09 INFO::MAIN :: Checking for shows with tvrage id's, since tvrage is gone
20:34:09 WARNING::MAIN :: Found 333 shows with TVRage ID', FIXING!
20:34:09 INFO::MAIN :: Mapping Space Battleship Yamato 2199 to tvdb id 259675
20:34:09 ERROR::MAIN :: Fatal error executing query: UNIQUE constraint failed: tv_shows.indexer_id
Traceback (most recent call last):
  File "/opt/SickBeard-tvrage/sickbeard/db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/opt/SickBeard-tvrage/sickbeard/db.py", line 88, in execute
    return self._execute(query, args)
  File "/opt/SickBeard-tvrage/sickbeard/db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
IntegrityError: UNIQUE constraint failed: tv_shows.indexer_id
Traceback (most recent call last):
  File "/opt/SickBeard-tvrage/SickBeard.py", line 522, in <module>
    SickRage().start()
  File "/opt/SickBeard-tvrage/SickBeard.py", line 295, in start
    sickbeard.initialize(consoleLogging=self.consoleLogging)
  File "/opt/SickBeard-tvrage/sickbeard/__init__.py", line 1308, in initialize
    db.sanityCheckDatabase(myDB, mainDB.MainSanityCheck)
  File "/opt/SickBeard-tvrage/sickbeard/db.py", line 254, in sanityCheckDatabase
    sanity_check(connection).check()
  File "/opt/SickBeard-tvrage/sickbeard/databases/mainDB.py", line 46, in check
    self.convert_tvrage_to_tvdb()
  File "/opt/SickBeard-tvrage/sickbeard/databases/mainDB.py", line 77, in convert_tvrage_to_tvdb
    (INDEXER_TVDB, mapping[0]['mindexer_id'], tvrage_show['indexer_id'])
  File "/opt/SickBeard-tvrage/sickbeard/db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/opt/SickBeard-tvrage/sickbeard/db.py", line 88, in execute
    return self._execute(query, args)
  File "/opt/SickBeard-tvrage/sickbeard/db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
sqlite3.IntegrityError: UNIQUE constraint failed: tv_shows.indexer_id

**Comment by faidwen** [comment]
Aaaah, okay, I wondered.  I thought the two had been separated etc.

So in actuality, in TVRAGE you have SHOWID1 and SHOWID2, but when converted to theTVDB they would in actuality have SHOWID3 (and not 2 separate ids). 

But do you actually care?  Aren't you simply just rolling all TVRAGE ids over to the TVDB ids?  So in the sql all SHOWID1 and SHOWID2 become SHOWID3 which corresponds to the TVDB id???

**Comment by miigotu** [comment]
The problem is that they are in different folders, which means they need a show object for each one, and you can only have one entry for a show id in the db. I could instruct SR to remove the second object entirely, but then you would be missing a show that you could not addExisting on.

I need to complicate the logic some, and do the moves in the check I think. I see no other way.

**Comment by mgaulton** [comment]
@miigotu so Find matches, merge folders with rename of second id to s02 maybe, remove secondary id sort of logic?

**Comment by miigotu** [comment]
@mgaulton yes, I need to do exactly that in the db sanity checks. Just wont be until later. Still no sleep.

**Comment by mgaulton** [comment]
@miigotu its cool, it'll keep man

**Comment by StevenChorkley** [comment]
As far as I know I didn't have any shows under TVRage yet my installation is scewed. 

I've literally only just updated too, perhaps this latest commit should be pulled until the fix is released?

**Comment by StevenChorkley** [comment]
https://i.imgur.com/O4Y4UCo.png

**Comment by OmgImAlexis** [comment]
@StevenChorkley we're not going to pull an update that's got over 150 commits in it. Only a select few are having issues and they should be resolved soon.

**Comment by faidwen** [comment]
@StevenChorkley  just rollback your GIT update manually, and turn off automatic updating until such time as there is a fix.

Back up and working.  And @miigotu needs sleep!!

**Comment by warplayer** [comment]
Mine is failing on the first show it tries to convert. Should I just continue to wait for an update later tonight? I'm on Windows Server and did a git pull on my Sickrage folder.

2015-09-08 18:08:29 WARNING  MAIN :: Found 6 shows with TVRage ID', FIXING!
2015-09-08 18:08:29 INFO     MAIN :: Mapping Humans to tvdb id 254367
2015-09-08 18:08:29 ERROR    MAIN :: Fatal error executing query: column indexer_id is not unique
Traceback (most recent call last):
  File "C:\SickRage\sickbeard\db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "C:\SickRage\sickbeard\db.py", line 88, in execute
    return self._execute(query, args)
  File "C:\SickRage\sickbeard\db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
IntegrityError: column indexer_id is not unique

**Comment by OmgImAlexis** [comment]
@warplayer yep, should be fixed as soon as @miigotu gets up and works on it.

**Comment by StevenChorkley** [comment]
@faidwen Which commit should I revert to?

**Comment by faidwen** [comment]
I went back to 

git checkout b2af183ad3bb07c8f416f83be96195425f2abd5d

Not sure if that is the best one, but it worked for me :)

**Comment by OmgImAlexis** [comment]
I think this is the commit before we merged lastnight 61043c08b570c23ba02724174ff3bc4b094f4462. Keep in mind you'll have a LOT of issues with the webui not working once it tries connecting to tvrage. This is why we pushed this updated out even though it was slightly broken.

**Comment by StevenChorkley** [comment]
@faidwen Worked perfectly

@OmgImAlexis I've never used tvrage, why would my installation fail?

**Comment by OmgImAlexis** [comment]
@StevenChorkley most of the issues people are having are from it. I assumed you had at least one show with it as almost everyone I've seen does. You should be okay then.

**Comment by StevenChorkley** [comment]
Commit 8f337c94896e32c79d36fb037ff98f9ecba5fd5a doesn't cause me startup errors, what caused this mess?

**Comment by mgaulton** [comment]
@StevenChorkley tvrage died i believe, this caused issues due to all the hooks in the code

**Comment by StevenChorkley** [comment]
@OmgImAlexis I'm assuming the issue was with commits d25bf5e9f503be6544befa7843bcc307f410a8af & 0577f675a430e2ba2ed0a726af7bf5f9cdfac56d attempting to reset the tvrage ID's? Should those two be removed for now? Otherwise your gonna have more and more users receiveing errors.

**Comment by JinkLeft** [comment]
I have the same error steven had. 

@StevenChorkley what command did you run to rollback to that commit. I'm a git noob.

**Comment by flightlevel** [comment]
@StevenChorkley 
It's not affecting everyone (no problems for me) and the issue is currently being worked on.

**Comment by StevenChorkley** [comment]
@ JinkLeft The last working commit is 08eaa7f743a174b87939d7b54398c18a9e574375

To return on linux, navigate to the sickrage folder and run `git reset --hard 08eaa7f743a174b87939d7b54398c18a9e574375`

**Comment by miigotu** [comment]
@StevenChorkley are you even reading the thread?
Read from here: https://github.com/SiCKRAGETV/sickrage-issues/issues/2693#issuecomment-138739499

Nothing is getting reverted as it is WORSE without it. 

If you cant deal with it for a few hours while I try to sleep then thats a little ridiculous.

**Comment by faidwen** [comment]
@miigotu aren't you supposed to be ASLEEP ;)  LOL!!!  

Thank you sir!   SLEEEEEEEEEEEEEP

**Comment by StevenChorkley** [comment]
@miigotu no,no, I can read thanks. My suggestion is to check if a user even used TVRage and then attempt to fix the ID's.

I also have an issue with the commit a298bb85351f1a868a78caaf81aa10319c1d0a5f along with others.... which was released before your fix.

I've no issue, I've rolled back, others won't.... so it was merely a suggestion.

**Comment by miigotu** [comment]
"My suggestion is to check if a user even used TVRage and then attempt to fix the ID's."
Which is exactly what the commits do.

Locking this until I wake up, so that other users seeing this problem don't come here and scroll down the the bottom looking for the latest news get a bunch of "Me too" "Im having this problem too"

For anyone coming in, the explanation of the problem starts around https://github.com/SiCKRAGETV/sickrage-issues/issues/2693#issuecomment-138739499

and continues through the next 5 or so posts.

**Comment by miigotu** [comment]
Temp fix: https://github.com/SiCKRAGETV/SickRage/commit/378f9818cb06867f777fa06a5bb03fba469b8ebd

It doesnt fix anytthing except boot.

**Comment by miigotu** [comment]
Thinking of a good way to merge the folders, its a pain in the ass.

**Comment by flightlevel** [comment]
@miigotu Could you just notify the user "TVRage has closed down. Due to a conflict between the way TVRage and TVDB have structured the episodes of this show please remove 'TvRageShowName' from SickRage."
Let the user move the files. There really shouldn't be that many for each person. None here and I had a few tvrage shows.

**Comment by miigotu** [comment]
I can just remove the dupe show from sickrage NP. Its the merging that will be too complicated for end users.

```
Jim Henson's The Storyteller/
    Season 01/
        S01EXX - *

Jim Henson's The Storyteller: Greek Myths/
    Season 02/
        S02EXX - *

```

These are both the same show, and season 2 of the second one needs moved into the first one.

**Comment by madeddie** [comment]
I've just left the files in the same location and changed the series id and season. Maybe not the cleanest of solutions, but it also saves me from cleaning up in Kodi.

**Comment by miigotu** [comment]
@madeddie have fun with dailySearch, backlog, show updates, postProcessing, etc.

**Comment by madeddie** [comment]
Yeah, for active/incomplete series this won't work indeed.

**Comment by neoatomic** [comment]
@miigotu After 3 day's being awake, you are probably running low on energy drinks.? (Sure the users will find there way to the Donate button. :p )

To be honest i dont think that a good fix can me created here...
I was thinking more along the lines, like let the user choose the option he wants to preform.
If TVRage shows are detected give a pop-up and let the user choose if he want to auto convert (with risk) or let SR auto remove the show from the DB.

**Comment by StevenChorkley** [comment]
I totally agree with @neoatomic
If a user decides to remove a show all s/he needs to do is re-add it,
Sickrage with redownload the rest! That's what it's designed to do after
all!

£10 donated from me :)

**Comment by miigotu** [comment]
It wont be able to be re-added directly, as you already have that show in your list =P. It would need moved to PP folder and processed (I was thinking to do just that)

If I just remove one of the shows from SR, the one which stays and is converted would then go and try to start downloading that missing season. This is a very pain in the ass issue because tvrage made a bad decision about shows which have "named seasons".

**Comment by flightlevel** [comment]
Pause any of the affected shows and let user know

**Comment by miigotu** [comment]
Can someone with this issue do some tests on my fix for this?

BACKUP YOUR DB!
Stop SR.
```bash
git fetch origin
git checkout -b tvrage-conversion-logic origin/tvrage-conversion-logic
```

Make sure to read any warnings or errors in the logs, and let me know if anything fails or needs better logging.

**Comment by MDamon** [comment]
@miigotu 

Starting SickRage
Traceback (most recent call last):
  File "SickBeard.py", line 522, in <module>
    SickRage().start()
  File "SickBeard.py", line 295, in start
    sickbeard.initialize(consoleLogging=self.consoleLogging)
  File "/home/<USER>/.sickrage/sickbeard/__init__.py", line 1308, in initialize
    db.sanityCheckDatabase(myDB, mainDB.MainSanityCheck)
  File "/home/<USER>/.sickrage/sickbeard/db.py", line 254, in sanityCheckDatabase
    sanity_check(connection).check()
  File "/home/<USER>/.sickrage/sickbeard/databases/mainDB.py", line 46, in check
    self.convert_tvrage_to_tvdb()
  File "/home/<USER>/.sickrage/sickbeard/databases/mainDB.py", line 77, in convert_tvrage_to_tvdb
    (INDEXER_TVDB, duplicate[0]['indexer_id']))
  File "/home/<USER>/.sickrage/sickbeard/db.py", line 168, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/<USER>/.sickrage/sickbeard/db.py", line 88, in execute
    return self._execute(query, args)
  File "/home/<USER>/.sickrage/sickbeard/db.py", line 76, in _execute
    return self.connection.cursor().execute(query)
sqlite3.OperationalError: near "WHERE": syntax error

**Comment by miigotu** [comment]
grr, one sec, Ill fix that

**Comment by miigotu** [comment]
@MDamon git pull, errant comma.

**Comment by MDamon** [comment]
@miigotu looks all fixed to me thanks!!  Are you submitting this to master at some point so I can get back on the gravy train?

**Comment by miigotu** [comment]
@MDamon yes, but I need to know what showed up in your log, can you post it on gist and gimme a link please? I want to make sure not to break anyone's DB.

**Comment by MDamon** [comment]
@miigotu sorry missed that req.. what log file are you looking for?

**Comment by MDamon** [comment]
@miigotu 
http://pastebin.com/w3eFSwgW

**Comment by miigotu** [comment]
Perfect. You did read this part right?

```
2015-09-09 09:09:15 INFO     MAIN :: Checking for shows with tvrage id's, since tvrage is gone
2015-09-09 09:09:15 WARNING  MAIN :: Found 1 shows with TVRage ID's, attempting automatic conversion...
2015-09-09 09:09:15 INFO     MAIN :: Processing Wet Hot American Summer: First Day of Camp at /media/net/Bin/TV/Wet Hot American Summer First Day of Camp
2015-09-09 09:09:15 INFO     MAIN :: Checking if there is already a show with id:%i in the show list
2015-09-09 09:09:15 WARNING  MAIN :: Found Wet Hot American Summer which has the same id as Wet Hot American Summer: First Day of Camp, cannot convert automatically so I am pausing Wet Hot American Summer
2015-09-09 09:09:15 WARNING  MAIN :: Removing Wet Hot American Summer: First Day of Camp and it's episodes from the DB
2015-09-09 09:09:15 WARNING  MAIN :: Manually move the season folders from /media/net/Bin/TV/Wet Hot American Summer First Day of Camp into /media/net/Bin/TV/Wet Hot American Summer, and delete /media/net/Bin/TV/Wet Hot American Summer First Day of Camp before rescanning Wet Hot American Summer and unpausing it
```

**Comment by MDamon** [comment]
@miigotu Guessing that was the issue eh..  Thanks again for the help!

**Comment by OmgImAlexis** [comment]
@miigotu, maybe add a popup notification letting them know to check the log for instructions?

**Comment by miigotu** [comment]
I was most interested in getting them up and running for now. I see this being a long-haul of mass issue spam on this one.

Not sure how to do the notification div, haven't really looked into that and was hoping to take a little break today. Kids wake up in 20 mins x.X

**Comment by mgaulton** [comment]
@miigotu I have an interesting twist, i have  '/mnt/anime' for guess what, anime and /mnt/tv/@@Sorted for regular shows. This logic wants to move converted anime from the anime folder to the regular folder, not a huge issue, just thought i'd point that out.

Logs/sickrage.log.1:2015-09-09 10:47:39 WARNING  MAIN :: Manually move the season folders from /mnt/anime/Space Battleship Yamato 2199 into /mnt/tv/@@Sorted/Star Blazers 2199, and delete /mnt/anime/Space Battleship Yamato 2199 before rescanning Star Blazers 2199 and unpausing it

**Comment by mgaulton** [comment]
@miigotu also, this is a miss. NCIS original and merging with NCIS:New Orleans

Logs/sickrage.log.1:2015-09-09 10:47:45 WARNING  MAIN :: Manually move the season folders from /mnt/tv/@@Sorted/Ncis into /mnt/tv/@@Sorted/NCIS New Orleans, and delete /mnt/tv/@@Sorted/Ncis before rescanning NCIS: New Orleans and unpausing it

**Comment by kerbysj** [comment]
Hi guys i know your all busy trying to fix the problems!

I started fresh just carrying over my config (good enough reason to clear out some shows that had ended etc) now i can even find any new episodes!

getting below error
SEARCHQUEUE-MANUAL-259393 :: [IPTorrents] :: Error while searching IPTorrents, skipping: maximum recursion depth exceeded

**Comment by miigotu** [comment]
If they are wanting to merge the wrong shows together then use add existing to add the TVRage show back using TVDB, as the mapping found in your DB was wrong to begin with.

@kerbysj we are working on a fix for that, separate issue.

**Comment by syrys** [comment]
if we are having merging issues, like what @mgaulton mentioned, and if automated conversion/merging is hard, then how about implementing something similar to this:

1. Disable all TVRage shows, make it obvious that they are disabled.
2. have an option to Enable each show by re-selecting/searching the show on the TVDB list (let the user search and select).
3. once selected, then do the merging process, atleast at this point you are most likely going to know that they selected the correct show.

This is probably time consuming to users, but heck, better that than loosing their data or incorrect merging.

**Comment by miigotu** [comment]
The posters arent broke. You just have images for the show with the old ID. Do a mass update on all shows, and after they are finished in the queue the images will be back.

**Comment by warplayer** [comment]
Sorry for the lame question, but I'm a little confused by some of the posts on here today.

So to get my Sickrage working, I need to do a git pull to get the most recent code, launch Sickrage, then Force Update all my shows? 

I tried looking through my log for a similar message like MDamon got that instructed him to move certain files, but didn't see anything. I did see a section that said 6 of my shows were on TV Rage. I have no clue how to find which ones were on Sickrage (I have over 400 shows in my library).

Thanks again for the hard work on getting this fix out quick!

**Comment by miigotu** [comment]
Stop SR, delete the cache folder, Start SR, Mass Update.

About the log lines. Just after the info about how many TVRage shows it found, if you didn't see any instructions about manually moving files/folders, then SR handled the conversion for you automatically.

There may or may not be an issue with showing poster images on Windows. This is a separate issue, that @MGaetan89 and @OmgImAlexis are working on.

**Comment by warplayer** [comment]
Ah, I skipped the step of deleting the cache folder. Thanks for that.

So, I had already started my Force Update. I just shut down the server, am going to delete the cache, will restart the Force Update. Do you think this will give me any problems or should I be good from there?

**Comment by miigotu** [comment]
it will be fine.

**Comment by aeror1** [comment]
@miigotu is the separate Windows issue that you mentioned with posters that @MGaetan89 and @OmgImAlexis are working on also the cause of very slow load time of the SR homepage (show list)?  

after the crash from the update, im up and running but with no posters and slow load time.  BRANCH: (master) / COMMIT: (ecaed9c745dfc645c4e26a268d1ddeca43b006ca) 

thanks!

**Comment by miigotu** [comment]
Try opening gui/slick/views/home.mako and replacing Lines 784-785

From:
```
        <% show_size = sickbeard.helpers.get_size(curShow._location) %>
        <td align="center" data-show-size="${show_size}">${sickbeard.helpers.pretty_filesize(show_size)}</td>
```
to
```
<td />
```

**Comment by aeror1** [comment]
@miigotu thank you sir!  its much gooder!  still no posters but you guys are aware of it.  thanks again.

**Comment by OmgImAlexis** [comment]
Load times should be fixed once this is merged. https://github.com/SiCKRAGETV/SickRage/pull/2470

**Comment by bartley69** [comment]
Arggg tvrage going down has butchered American Dad! It started downloading over my eps. The tvdb has fucked American Dad up :-(

**Comment by miigotu** [comment]
@bartley69 not really, you just unpaused the show without first following directions maybe? Perform a rescan on the show.

**Comment by bartley69** [comment]
American Dad numbering from TVDB is completely different from the Scene numbering. TV Rage followed the scene numbering but now that is gone it has reverted back to the TVDB and screwed up the season and episode numbering. It's hard to explain i guess....But sickrage uses scene numbering for downloading, no new eps will be downloaded anymore. Here are some pics of my Library now with the show using The TVBD numbering when before it was perfect...

![ad1](https://cloud.githubusercontent.com/assets/10230287/9829617/7d329190-593e-11e5-8425-0a97842afacb.png)
![ad2](https://cloud.githubusercontent.com/assets/10230287/9829619/83ee82dc-593e-11e5-902e-22c0751c4914.png)

**Comment by miigotu** [comment]
edit the show, enable scene numbering, and force a full update.

**Comment by bartley69** [comment]
It is already ticked, nothing changed. Seasons are still "wrong." TVDB follows the order and everyone else follows another, it's messed up. Don't know how to fix it, but TV Rage used to "right" ordering.

**Comment by miigotu** [comment]
for scene numbering we don't use tvdb numbering. we use the mapping on thexem.de

**Comment by miigotu** [comment]
also, these comments have no place in this closed issue. as it has nothing to do with SR failing to start.
